design-proposals/app-orch-dp-import.md

```diff
@@ -0,0 +1,160 @@
+the repository to a temporary section of its local file system and use file operations to examine
+the repository.
+
+A more sophisticated approach may be use github REST APIs to examine the contents of a repository
+without having to fully clone the repo. As the repositories may be large and there are only a few
+relatively small yaml files that are of interest, this may be more efficient.
+
+Note: The above URLs use query arguments for example illustrative purposes. The implementation may use POST
+with a JSON payload instead. This is an implementation detail.
+
+### Proposal 3: Import all deployment packages from a repository (requires less GUI work)
+
+This is the same as the preceding proposal, but with a simplification that it elminates the step required
+to generate list of DPs and for the user to select a DP from the list.
+
+- The existing `Import Deployment Package
+  <https://docs.openedgeplatform.intel.com/edge-manage-docs/main/user_guide/package_software/import_deployment.html>`
+  screen shall be updated with an `Import from Github Repository` button.
+
+- The user will paste the URL of the repo they wish to import into the GUI.
+
+- The GUI will pass the `URL` to the catalog service
+  for import, for example `/v3/projects/{project-name}/catalog/import_package_from_repo?url=<url>`.
+
+- The application catalog service shall extract all deployment packages from the repository and
+  add all of them to the application catalog.
+
+- The GUI shall display a confirmation screen.
+
+## Rationale
+
+Other approaches have been considered include:
+
+- Requiring the user to clone the repo to their computer and import through their browser.
+  This is the current solution. Requiring the user to clone to their computer requires additional
+  tools and skills, consumes resources on their computer, and presents additional user experience
+  burden.
+
+- An API between Edge Software Catalog and Orchestrator could be created to allow one service to push to
+  (or pull from) the other. This would involve additional complexity regarding authentication and
+  entitlement from ESC to Orchestrator or vice-versa. It would require additional implementation effort
+  from the ESC.
+
+- The App Orch Tenant Controller (AOTC) is capable of installing deployment packages for extensions and could be used to
+  install application deployment packages. However, the AOTC is a one-size-fits-all solution that bootstraps a project
+  with a predefined set of packages. It is not designed to install new packages at runtime. There may be 50 or
+  more AI applications with published deployment packages, and it would be undesirable to preload that many into
+  the catalog at project creation time. Similarly, there may be a need to import deployment packages at any
+  time.
+
+## Affected components and Teams
+
+- Application Orchestration Application Catalog.
+
+- GUI.
+
+- CLI.
+
+## Implementation plan
```

Comment:
This only applies to approach #2, regardless do we need this level of implementation details at this ADR level?

Reply:
`Implementation Plan` is part of the template. Mostly I wanted to clarify the split between UI and backend responsibilities. Let me clarify approach 1 vs 2/3.

Reply:
updated.